### PR TITLE
fix: Update the JSON-RPC to correct the return information.

### PR DESCRIPTION
**Pull request by WilliamXieCrypto**
Fixes: #632 

Solution:

Before the fix, the "web3_sha3" JSON_RPC endpoint will overwrite the "0x" as "" by using "hexutil.Bytes" as the input parameters, then return same result. After the fix, using "string" as the input parameters will avoid the overwrite and return correct response.

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: tests/rpc/rpc_test.go [hunk 1]
```diff
@@ -135,6 +135,33 @@ func callWithError(method string, params interface{}) (*Response, error) {
 	return rpcRes, nil
 }
 
+func TestWeb3_Sha3(t *testing.T) {
```
Comment (fedekunze) [inline tests/rpc/rpc_test.go:138]: can you move the tests to this file and use the client RPC instead?
https://github.com/tharsis/ethermint/blob/main/tests/e2e/integration_test.go

**Review by fedekunze — COMMENTED** [review]
Thanks! minor comment to move the tests to the new package

File: tests/rpc/rpc_test.go (hunk 1 above)
Comment (WilliamXieCrypto) [reply tests/rpc/rpc_test.go]: @fedekunze sure, updated

File: tests/e2e/integration_test.go [hunk 2]
```diff
@@ -685,3 +687,26 @@ func (s *IntegrationTestSuite) waitForTransaction() {
 func TestIntegrationTestSuite(t *testing.T) {
 	suite.Run(t, new(IntegrationTestSuite))
 }
+
+func (s *IntegrationTestSuite) TestWeb3Sha3() {
```
Comment (fedekunze) [inline tests/e2e/integration_test.go:691]: can you refactor the tests and use table tests instead?

Comment (WilliamXieCrypto) [reply tests/e2e/integration_test.go:691]: @fedekunze done, thanks for your advice

**Review by fedekunze — APPROVED** [review]

**Pull request merged**